pkg/controller/certificates/certificate_request.go

```diff
@@ -483,6 +483,44 @@ func (c *certificateRequestManager) processCertificate(ctx context.Context, crt
 		}
 	}
 
+	// If the CertificateRequest exists but has failed then we check the failure
+	// time. If the failure time doesn't exist or is over an hour in the past
+	// then delete the request so it can be re-created on the next sync. If the
+	// failure time is less than an hour in the past then schedule this owning
+	// Certificate for a re-sync in an hour.
+	if existingReq != nil && apiutil.CertificateRequestHasFailed(existingReq) {
+		if existingReq.Status.FailureTime == nil {
+			log.Info("the failed existing certificate request has no failure time so will be retried now")
+
+		} else if c.clock.Since(existingReq.Status.FailureTime.Time) > time.Hour {
+			log.Info("the failed existing certificate request failed over an hour ago so will be retried now")
+
+		} else {
+			log.Info("the failed existing certificate request failed less than an hour ago, will be scheduled for reprocessing in an hour")
+
+			key, err := keyFunc(crt)
+			if err != nil {
+				log.Error(err, "error getting key for certificate resource")
+				return nil
+			}
+
+			c.scheduledWorkQueue.Add(key, time.Hour)
+
+			c.recorder.Eventf(crt, corev1.EventTypeNormal, "CertificateRequestReschedule", "The CertificateRequest %q has failed and is scheduled for a retry in 1 hour", existingReq.Name)
```

Comment:
We shouldn't fire an Event here - this block could be called multiple times in quick succession if a renewal is scheduled at the same time the status is updated (or even if the Secret resource has a new label added etc.)

Reply:
To clarify: informing the user *is* desired, but I think we need to consider how we do it.

Once we have more advanced back-off mechanisms, we may also be able to record e.g. `nextRetryTime` on the Certificate resource, and only fire an Event if this is `nil` and we have just set it.